networkx/algorithms/community/louvain.py

```diff
@@ -0,0 +1,289 @@
+"""Function for detecting communities base on Louvain Community Detection
+Algorithm"""
+
+from collections import deque, defaultdict
+
+import networkx as nx
+from networkx.algorithms.community import modularity
+from networkx.utils import py_random_state
+
+__all__ = ["louvain_communities", "generate_dendrogram"]
+
+
+@py_random_state("seed")
+def louvain_communities(
+    G, weight="weight", resolution=1, threshold=0.0000001, seed=None
+):
+    r"""Find the best partition of G using the Louvain Community Detection
+    Algorithm.
+
```

Comment:
Maybe add a first paragraph description of what Louvain tries to do, with citation to the papers [1] and [2].
A similar description (maybe the exact wording so readers know they've already read it) could go into `generate_dendogram`.